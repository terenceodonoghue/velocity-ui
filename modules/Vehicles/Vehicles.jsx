/** @jsx jsx */
import { jsx } from '@emotion/core';
import * as S from './Vehicles.styles';

const Table = () => (
  <S.Table />
);

export default {
  Table,
};
