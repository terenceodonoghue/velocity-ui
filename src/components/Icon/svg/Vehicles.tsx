import React from 'react';
import { IconProps } from '../Icon';

const SvgVehicles: React.FunctionComponent<IconProps> = ({
  fill,
  theme,
  title,
  ...props
}) => (
  <svg viewBox="0 0 24 24" fill="none" {...props}>
    {title ? <title>{title}</title> : null}
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M21.428 8.732L18.602 4.73C18.269 4.256 17.831 4 17.378 4h-6.303c-.498 0-.97.309-1.309.859L7.378 8.732c-.15.249-.363.384-.582.392h-2.14C3.19 9.116 2 10.917 2 13.132v1.575c0 1.492.801 2.705 1.791 2.705h.269c.358 1.899 1.661 2.992 2.915 2.449.781-.34 1.393-1.266 1.617-2.449h6.836c.358 1.899 1.662 2.992 2.915 2.449.781-.34 1.393-1.266 1.617-2.449h.244c.985.008 1.791-1.206 1.796-2.697v-4.001c-.005-.754-.214-1.47-.572-1.982zm-4.06-3.225c.21 0 .408.12.552.332l2.314 3.285h-2.727l-1.104-3.617h.965zm-.95 3.617h-2.781V5.507h1.676l1.105 3.617zm-10.1 9.38c-.75 0-1.363-.926-1.363-2.064s.612-2.065 1.363-2.065c.752 0 1.364.927 1.364 2.065 0 1.145-.612 2.065-1.364 2.065zm6.324-2.606h-3.99c-.19-1.952-1.388-3.293-2.677-3.007-1.025.234-1.83 1.454-1.985 3.007h-.199c-.438.007-.796-.528-.8-1.19v-1.576c-.001-1.386.745-2.509 1.66-2.509h2.145c.035 0 .065 0 .1-.007.02 0 .034.007.054.007h5.692v5.275zm0-6.774h-4.14l1.99-3.225c.15-.249.364-.384.583-.392h1.567v3.617zM15.726 4h-3.035 3.035zm1.966 14.505c-.752 0-1.364-.927-1.364-2.065s.612-2.065 1.364-2.065c.75 0 1.363.927 1.363 2.065 0 1.145-.612 2.065-1.363 2.065zm3.313-3.798c-.005.663-.363 1.198-.796 1.198h-.184c-.19-1.951-1.388-3.3-2.677-3.006-1.025.233-1.83 1.454-1.985 3.006h-1.726v-5.274h7.278c.03 0 .055-.008.085-.015 0 .037.005.068.005.105v3.986z"
      fill={fill || theme.colors.lightBlueGrey}
    />
    <mask
      id="Vehicles_svg__a"
      maskUnits="userSpaceOnUse"
      x={2}
      y={4}
      width={20}
      height={16}
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M21.428 8.732L18.602 4.73C18.269 4.256 17.831 4 17.378 4h-6.303c-.498 0-.97.309-1.309.859L7.378 8.732c-.15.249-.363.384-.582.392h-2.14C3.19 9.116 2 10.917 2 13.132v1.575c0 1.492.801 2.705 1.791 2.705h.269c.358 1.899 1.661 2.992 2.915 2.449.781-.34 1.393-1.266 1.617-2.449h6.836c.358 1.899 1.662 2.992 2.915 2.449.781-.34 1.393-1.266 1.617-2.449h.244c.985.008 1.791-1.206 1.796-2.697v-4.001c-.005-.754-.214-1.47-.572-1.982zm-4.06-3.225c.21 0 .408.12.552.332l2.314 3.285h-2.727l-1.104-3.617h.965zm-.95 3.617h-2.781V5.507h1.676l1.105 3.617zm-10.1 9.38c-.75 0-1.363-.926-1.363-2.064s.612-2.065 1.363-2.065c.752 0 1.364.927 1.364 2.065 0 1.145-.612 2.065-1.364 2.065zm6.324-2.606h-3.99c-.19-1.952-1.388-3.293-2.677-3.007-1.025.234-1.83 1.454-1.985 3.007h-.199c-.438.007-.796-.528-.8-1.19v-1.576c-.001-1.386.745-2.509 1.66-2.509h2.145c.035 0 .065 0 .1-.007.02 0 .034.007.054.007h5.692v5.275zm0-6.774h-4.14l1.99-3.225c.15-.249.364-.384.583-.392h1.567v3.617zM15.726 4h-3.035 3.035zm1.966 14.505c-.752 0-1.364-.927-1.364-2.065s.612-2.065 1.364-2.065c.75 0 1.363.927 1.363 2.065 0 1.145-.612 2.065-1.363 2.065zm3.313-3.798c-.005.663-.363 1.198-.796 1.198h-.184c-.19-1.951-1.388-3.3-2.677-3.006-1.025.233-1.83 1.454-1.985 3.006h-1.726v-5.274h7.278c.03 0 .055-.008.085-.015 0 .037.005.068.005.105v3.986z"
        fill="#fff"
      />
    </mask>
  </svg>
);

export default SvgVehicles;
