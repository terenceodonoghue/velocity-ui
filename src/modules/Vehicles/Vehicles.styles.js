import styled from '@emotion/styled';
import { Container, Global, Input } from '../../components';

const {
  variables: { colors, typography },
} = Global;

export const Filter = styled(Container.Card)`
  padding: 32px 24px 3px;
  width: 268px;
`;

export const FilterField = styled(Input.Text)`
  padding: 0 6px 19px;
  color: ${colors.dark};
  font-size: 15px;
  font-weight: ${typography.fontWeightRegular};
  letter-spacing: normal;
  line-height: 1.47;
  text-transform: none;

  input {
    color: ${colors.clearBlue};
  }
`;

export const FilterLabel = styled.span`
  flex: 1;
`;

export const FilterText = styled.div`
  display: flex;
`;

export const FilterValue = styled.span`
  color: ${colors.blueyGrey};
`;

export const SliderInput = styled(Input.Slider)`
  margin: 12px 0 18px;
`;

export const TextInput = styled(Input.Text)`
  background: ${colors.white};
  color: ${colors.clearBlue};
`;

export const Vehicles = styled.table``;
