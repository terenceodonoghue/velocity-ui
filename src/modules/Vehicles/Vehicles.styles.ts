import { css, InterpolationWithTheme } from '@emotion/core';
import { Theme } from '../../components';

export const filter = css`
  padding: 32px 24px 3px;
  width: 268px;
`;

export const sliderInput = css`
  margin: 0 6px 34px;
`;

export const textInput: InterpolationWithTheme<Theme> = (theme) => css`
  margin: 0 0 19px;
  color: ${theme.colors.black};
  font-size: 15px;
  font-weight: ${theme.typography.fontWeightRegular};
  letter-spacing: normal;
  line-height: 1.47;
  text-transform: none;

  input {
    color: ${theme.colors.blue};
  }
`;
