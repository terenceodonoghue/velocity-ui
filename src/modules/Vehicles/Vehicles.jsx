/** @jsx jsx */
import { jsx } from '@emotion/core';
import * as S from './Vehicles.styles';

const Filter = () => (
  <S.Filter heading="Filter">
    <S.FilterField>
      <S.FilterText>
        <S.FilterLabel>Trips taken</S.FilterLabel>
        <S.FilterValue>753</S.FilterValue>
      </S.FilterText>
      <S.SliderInput />
    </S.FilterField>
    <S.FilterField>
      <S.FilterText>
        <S.FilterLabel>Service due</S.FilterLabel>
        <S.FilterValue>14 days</S.FilterValue>
      </S.FilterText>
      <S.SliderInput />
    </S.FilterField>
    <S.FilterField>
      Vehicle model
      <S.TextInput placeholder="Vehicle model" />
    </S.FilterField>
    <S.FilterField>
      Status
      <S.TextInput placeholder="Status" value="Service needed" />
    </S.FilterField>
    <S.FilterField>
      Location
      <S.TextInput placeholder="Location" value="USA" />
    </S.FilterField>
  </S.Filter>
);

const Vehicles = () => (
  <S.Vehicles />
);

export default {
  Filter,
  Vehicles,
};
