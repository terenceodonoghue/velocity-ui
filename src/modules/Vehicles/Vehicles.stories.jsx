import React from 'react';
import { storiesOf } from '@storybook/react';
import { Filter, Vehicles } from './Vehicles';

storiesOf('Modules/Vehicles Dashboard', module)
  .add('Filter', () => <Filter />)
  .add('Vehicles', () => <Vehicles />);
