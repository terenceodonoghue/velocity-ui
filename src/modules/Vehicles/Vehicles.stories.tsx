import React from 'react';
import { storiesOf } from '@storybook/react';
import { Filter } from './Vehicles';

storiesOf('Modules/Vehicles Dashboard', module).add('Filter', () => <Filter />);
