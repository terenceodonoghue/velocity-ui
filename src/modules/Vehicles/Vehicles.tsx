/* eslint-disable import/prefer-default-export */
/** @jsx jsx */
import { jsx } from '@emotion/core';
import { FunctionComponent } from 'react';
import { Inputs, Surfaces } from '~/components';
import * as css from './Vehicles.styles';

export const Filter: FunctionComponent = () => (
  <Surfaces.Card css={css.filter} heading="Filter">
    <Inputs.Slider
      css={css.sliderInput}
      primary="Trips taken"
      secondary="753"
    />
    <Inputs.Slider
      css={css.sliderInput}
      primary="Service due"
      secondary="14 days"
    />
    <Inputs.Text
      css={css.textInput}
      inputProps={{
        placeholder: 'Vehicle model',
        type: 'text',
      }}
      type="text"
    >
      Vehicle model
    </Inputs.Text>
    <Inputs.Text
      css={css.textInput}
      inputProps={{
        defaultValue: 'Service needed',
        placeholder: 'Status',
        type: 'text',
      }}
    >
      Status
    </Inputs.Text>
    <Inputs.Text
      css={css.textInput}
      inputProps={{
        defaultValue: 'USA',
        placeholder: 'Location',
        type: 'text',
      }}
    >
      Location
    </Inputs.Text>
  </Surfaces.Card>
);
