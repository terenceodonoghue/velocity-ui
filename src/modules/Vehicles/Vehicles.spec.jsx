import React from 'react';
import Vehicles from './Vehicles';

describe('Vehicles', () => {
  describe('Filter', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(
        <Vehicles.Filter />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Vehicles', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(
        <Vehicles.Vehicles />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
