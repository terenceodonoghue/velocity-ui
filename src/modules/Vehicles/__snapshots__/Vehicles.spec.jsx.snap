// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles Filter matches snapshot 1`] = `
.emotion-18 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: #fff;
  padding: 32px 24px 3px;
  width: 268px;
}

.emotion-0 {
  margin: 0 0 26px 0;
  color: #8798ad;
  font-size: 13px;
  font-weight: 400;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

.emotion-5 {
  display: inline-block;
  color: #b0bac9;
  font-size: 12px;
  font-weight: 500;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
  padding: 0 6px 19px;
  color: #2e384d;
  font-size: 15px;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.47;
  text-transform: none;
}

.emotion-5 input {
  color: #2e5bff;
}

.emotion-2 {
  display: block;
  border: solid 1px #e0e7ff;
  border-radius: 5px;
  height: 38px;
  margin: 7px 0;
  max-width: 270px;
  padding: 10px 16px;
  width: 100%;
  background-color: rgba(224,231,255,0.2);
  color: #2e384d;
  font-family: inherit;
  font-size: 15px;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.emotion-2:disabled {
  opacity: 0.3;
}

.emotion-2:focus {
  border: solid 1px #2e5bff;
  outline: 0;
}

<div
  className="emotion-17 emotion-18 emotion-19"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Filter
  </h3>
  <label
    className="emotion-4 emotion-5 emotion-6"
  >
    Vehicle model
    <input
      className="emotion-2 emotion-3"
      placeholder="Vehicle model"
      type="text"
    />
  </label>
  <label
    className="emotion-4 emotion-5 emotion-6"
  >
    Status
    <input
      className="emotion-2 emotion-3"
      defaultValue="Service needed"
      placeholder="Status"
      type="text"
    />
  </label>
  <label
    className="emotion-4 emotion-5 emotion-6"
  >
    Location
    <input
      className="emotion-2 emotion-3"
      defaultValue="USA"
      placeholder="Location"
      type="text"
    />
  </label>
</div>
`;

exports[`Vehicles Vehicles matches snapshot 1`] = `
<table
  className="emotion-0 emotion-1"
/>
`;
